modules/azure/compute.go

```diff
@@ -8,97 +8,392 @@ import (
+// VirtualMachineExists indicates whether the specifcied Azure Virtual Machine exists.
+// This function would fail the test if there is an error.
+func VirtualMachineExists(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) bool {
+	exists, err := VirtualMachineExistsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// VirtualMachineExistsE indicates whether the specifcied Azure Virtual Machine exists.
+func VirtualMachineExistsE(vmName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get VM Object
+	_, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetVirtualMachineNics gets a list of Network Interface names for a specifcied Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineNics(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	nicList, err := GetVirtualMachineNicsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return nicList
+}
+
+// GetVirtualMachineNicsE gets a list of Network Interface names for a specified Azure Virtual Machine.
+func GetVirtualMachineNicsE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a VM client
-	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmNICs := *vm.NetworkProfile.NetworkInterfaces
+	if len(vmNICs) == 0 {
+		// No NIC present
+		return nil, nil
+	}
 
-	// Create an authorizer
-	authorizer, err := NewAuthorizer()
+	// Get the Names of the attached NICs
+	nics := make([]string, len(vmNICs))
+
+	for i, nic := range vmNICs {
+		nics[i] = GetNameFromResourceID(*nic.ID)
+	}
+	return nics, nil
+}
+
+// GetVirtualMachineManagedDisks gets the list of Managed Disk names of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineManagedDisks(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	diskNames, err := GetVirtualMachineManagedDisksE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return diskNames
+}
+
+// GetVirtualMachineManagedDisksE gets the list of Managed Disk names of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDisksE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Attach authorizer to the client
-	vmClient.Authorizer = *authorizer
+	// Get VM attached Disks
+	vmDisks := *vm.StorageProfile.DataDisks
```

Comment:
Same question on `nil`

Reply:
Similar reasoning, the OS disk is storage and will always have one. Added the extra check still. If you rather remove these nil checks I will do so, just exercising an abundance of caution to avoid future panics.